src/NuGet.Core/NuGet.Configuration/Utility/PathValidator.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text.RegularExpressions;
+using NuGet.Common;
+
+namespace NuGet.Configuration
+{
+    public static class PathValidator
+    {
+        private static readonly char[] _invalidPathChars = Path.GetInvalidPathChars();
+        /// <summary>
+        /// Validates that a source is a valid path or url.
+        /// </summary>
+        /// <param name="source">The path to validate.</param>
+        /// <returns>True if valid, False if invalid.</returns>
+        public static bool IsValidSource(string source)
+        {
+            return IsValidLocalPath(source) || IsValidUncPath(source) || IsValidUrl(source) || IsValidRelativePath(source);
+        }
+
+        /// <summary>
+        /// Validates that path is properly formatted as a local path. 
+        /// </summary>
+        /// <remarks>
+        /// On Windows, a valid local path must starts with the drive letter.
+        /// Example: C:\, C:\path, C:\path\to\
+        /// Bad: C:, C:\\path\\, C:\invalid\*\"\chars
+        /// </remarks>
+        /// <param name="path">The path to validate.</param>
+        /// <returns>True if valid, False if invalid.</returns>
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "We don't want to throw during detection")]
+        public static bool IsValidLocalPath(string path)
+        {
+            try
+            {
+                if (RuntimeEnvironmentHelper.IsWindows)
+                {
+                    // Checking driver letter on Windows
+                    if (!Regex.IsMatch(path.Trim(), @"^[A-Za-z]:\\"))
+                    {
+                        return false;
+                    }
+                }
+
+                return Path.IsPathRooted(path) && (path.IndexOfAny(_invalidPathChars) == -1);
+            }
+            catch
+            {
+                return false;
+            }
+
+        }
+
+        /// <summary>
+        /// Validates that path is properly formatted as an UNC path. 
+        /// </summary>
+        /// <remarks>
+        /// Example: \\server\share, \\server\share\path, \\server\share\path\to\
+        /// Bad: \\missingshare, \\server\invalid\*\"\chars
+        /// </remarks>
+        /// <param name="path">The path to validate.</param>
+        /// <returns>True if valid, False if invalid.</returns>
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "We don't want to throw during detection")]
+        public static bool IsValidUncPath(string path)
+        {
+            try
+            {
+                Path.GetFullPath(path);
+                return Regex.IsMatch(path.Trim(), @"^\\\\");
```

Comment:
Generally we should avoid using regex's. Especially in this case when 2-chars comparison would be straightforward.